dbt/adapters/snowflake/impl.py

```diff
@@ -131,6 +141,86 @@ def get_columns_in_relation(self, relation):
             else:
                 raise
 
+    def _show_object_metadata(self, relation: SnowflakeRelation) -> Optional[dict]:
+        try:
+            kwargs = {"relation": relation}
+            results = self.execute_macro(SHOW_OBJECT_METADATA_MACRO_NAME, kwargs=kwargs)
+
+            if len(results) == 0:
+                return None
+
+            return results
+        except DbtDatabaseError:
+            return None
+
+    def get_catalog_for_single_relation(
+        self, relation: SnowflakeRelation
+    ) -> Optional[CatalogTable]:
+        object_metadata = self._show_object_metadata(relation)
+
+        if not object_metadata:
+            return None
+
+        row = object_metadata[0]
+
+        is_dynamic = row.get("is_dynamic") in ("Y", "YES")
+        kind = row.get("kind")
+
+        if is_dynamic and kind == str(SnowflakeRelationType.Table).upper():
+            table_type = str(SnowflakeRelationType.DynamicTable).upper()
+        else:
+            table_type = kind
+
+        # https://docs.snowflake.com/en/sql-reference/sql/show-views#output
+        is_view = kind in (
+            str(SnowflakeRelationType.View).upper(),
+            str(SnowflakeRelationType.MaterializedView).upper(),
```

Comment:
Per my comment on `SnowflakeRelationType`, we don't need to check for `MaterializedView`.